workflow/executor/executor.go

```diff
@@ -169,6 +169,10 @@ func (we *WorkflowExecutor) LoadArtifacts(ctx context.Context) error {
 				return argoerrs.Errorf(argoerrs.CodeNotFound, "required artifact '%s' not supplied", art.Name)
 			}
 		}
+		err := art.CleanPath()
```

Comment:
I moved this check ahead of the call to `newDriverArt` since `newDriverArt` makes a deep clone of this artifact, so I wanted it to be operating on an artifact that has a clean path.